File: src/main/java/org/wendel/card/Card.java
package org.wendel.card;

import java.util.Objects;

public class Card {

    private final int number;
    private final Suit suit;

    public Card(int number, Suit suit) {
        this.number = number;
        this.suit = suit;
    }

    public int getNumber() {
        return number;
    }

    public Suit getSuit() {
        return suit;
    }

    @Override
    public String toString() {
        String prefix = number < 10 ? " " : "";
        return prefix + number + suit;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Card card = (Card) o;
        return number == card.number && suit == card.suit;
    }

    @Override
    public int hashCode() {
        return Objects.hash(number, suit);
    }
}

File: card/Card.cs
namespace card_trick.card;

public class Card
{
    private readonly int _value;
    private readonly Suit _suit;

    public Card(int value, Suit suit)
    {
        _value = value;
        _suit = suit;
    }
    
    public int Value => _value;
    public Suit Suit => _suit;

    public override string ToString()
    {
        var prefix = Value < 10 ? " " : "";
        return $"{prefix}{Value}{_suit.ToUnicodeString()}";
    }
}